src/app/personalised-homepage/page.js

```diff
@@ -22,6 +23,12 @@ function PersonalisedHomepage() {
     const [albums, setAlbums] = useState([]);
     const [tracks, setTracks] = useState([]);
     const [playlists, setPlaylists] = useState([]);
+    const [userTopArray, setUserTopArray] = useState([]);
+
+    // Variables / state used to control playback
+    const [playTrack, setPlayTrack] = useState("");
+    const [playerType, setPlayerType] = useState("");
+    const [playback, setPlayback] = useState(false);
```

Comment:
**Informational:** When handling React states, there is an argument to be had to reduce the amount of states we create. We can do this by grouping related states into a single object. 

Each time you call a state setter function, React may re-render your component. If you have many different pieces of state and update them frequently, this can lead to unnecessary re-renders and potentially impact performance. 

By reducing the number of states and batching related state updates together, you can potentially reduce the number of re-renders and improve performance. This mainly affects users with low-end/old devices. Modern devices would probably be unnoticeable.

### Combining related states
**State definition:**
```js
const [player, setPlayer] = useState({
  playTrack: "",
  playerType: "",
  playback: false
});
```

**State update:**
```js
function setPlaylistId(e) {
    setPlayer(prevState => ({
        ...prevState,
        playTrack: e.currentTarget.id,
        playerType: "playlist",
        playback: true
    }));
}
```